src/python/pants/backend/python/tasks/select_interpreter.py

```diff
@@ -44,7 +46,7 @@ class SelectInterpreter(Task):
   def implementation_version(cls):
     # TODO(John Sirois): Fixup this task to use VTS results_dirs. Right now version bumps aren't
```

Comment:
This TODO is relevant to this issue. But I think that it should be removed.

Because this task is unioning a bunch of different cache keys in order to come up with a single interpreter for all of them, it will not be able to actually use the "automatic caching" as described here: https://www.pantsbuild.org/dev_tasks.html#enabling-artifact-caching-for-tasks ... and it won't be able to use the results dirs. It's similar to an ivy or coursier resolve in that regard (...and probably to the pyprep tasks as well).

As an additional defense beyond the unioned target cache keys (which should be sufficient) it should probably write any data it produces under:
https://github.com/pantsbuild/pants/blob/36fdbd8973763e8b2dae994b25dd19f79cf20301/src/python/pants/task/task.py#L284-L295
...but that will only include the task fingerprint, so the unioned target keys are still necessary.

Reply:
(that won't be the whole task fingerprint though, including the fingerprints of transitively dependent subsystems, just the implementation version -- is there a reason that's sufficient here, or would we need to modify `self.versioned_workdir`?)

Reply:
There is some annoying redundancy in there, but I'm not sure it's worth tackling here.

The long multi-part path that is generated under `.pants.d` for "automatic caching" (linked above) is generated here:
https://github.com/pantsbuild/pants/blob/36fdbd8973763e8b2dae994b25dd19f79cf20301/src/python/pants/invalidation/cache_manager.py#L338-L350

...but that only actually kicks in when `Task.create_target_dirs` or `Task.cache_target_dirs` are set for automatic caching. It would be nice to move all of the task identity/fingerprinting stuff out of that codepath and into `Task` itself (to replace `self.workdir` and `self.versioned_workdir`)... but that would be good to do in a separate change. And I've waffled about touching it because I have hoped that we will be able to focus on v2 tasks more exclusively soon.